### PR TITLE
Issue #1677 - Allow users to define app specific urls to expose in the UI

### LINKED ISSUE #1677: Allow users to define app specific urls to expose in the UI
**Is your feature request related to a problem? Please describe.**
Users should be able to define a list of useful URLs for their specific application and expose those links on the UI. These links should provide helpful shortcuts that make easier to integrate Argo CD into existing systems by making it easier to find other components inside and outside Argo CD.  As an example, they could reference a CI pipeline or an application specific management tool.

**Describe the solution you'd like**
We can introduce a new struct to the Application CRD called `info` which has a name and value field. When the user clicks the application bubble in the UI, these name values pair are displayed in a new block below the Sync Policy block.

**Have you thought about contributing yourself?**
This would be a great first-time issue!

### PR DESCRIPTION
**Pull request by AdityaGupta1**
Closes #1677 

Allows users to create a list of "infos" - URLs, email addresses, and/or plain text - that shows up in the application summary window.

### CODE REVIEW

**Comment by alexec** [comment]
@dthomson25 I've cheekily assigned you as reviewed - but really you are the customer! Is this what you want?

**Review by alexec — COMMENTED** [review]
@AdityaGupta1, as a reviewer, my goal is to help you get your PR merged, 

- Plural of "info" is "info". Like the plural of "sheep" is "sheep".
- How do I edit this? Via the UI? 
- We should talk about tests and maybe adding a doc page.

**Comment by AdityaGupta1** [comment]
@alexec 

- I know the plural of "info" is just "info" but I made it "infos" to avoid any confusion between the list of info and a singular info object.
- This is currently only editable through the UI on the application creation and summary pages, but @dthomson25 suggested sending the PR with this much and making a separate issue for adding command line support.
- Sure, what should I do for that?

**Comment by dthomson25** [comment]
Yeah, this is the functionality that I was expecting! Once merged, we can start vending out apps with relevant links, which will reduce the amount of time finding them while debugging.

I agree with Alex on using info instead of infos since other people will assume it is a typo.

**Comment by AdityaGupta1** [comment]
@dthomson25 I renamed infos to info and moved the function out

**Review by jessesuen — DISMISSED** [review]

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -46,6 +46,8 @@ type ApplicationSpec struct {
 	SyncPolicy *SyncPolicy `json:"syncPolicy,omitempty" protobuf:"bytes,4,name=syncPolicy"`
 	// IgnoreDifferences controls resources fields which should be ignored during comparison
 	IgnoreDifferences []ResourceIgnoreDifferences `json:"ignoreDifferences,omitempty" protobuf:"bytes,5,name=ignoreDifferences"`
+	// Infos contains a list of useful information (URLs, email addresses, and plain text) that relates to the application
+	Infos []Info `json:"infos,omitempty" protobuf:"bytes,6,name=infos"`
```
[the remark this answers is not in the document]
Comment (jessesuen) [reply pkg/apis/application/v1alpha1/types.go]: I see this was already resolved. Ignore.

File: ui/src/app/applications/components/application-summary/application-summary.tsx [hunk 2]
```diff
@@ -107,6 +108,137 @@ export const ApplicationSummary = (props: {
         }
     }
 
+    class EditableInfoList extends React.Component<{}, { editing: boolean, saving: boolean }> {
```
[the remark this answers is not in the document]
Comment (AdityaGupta1) [reply ui/src/app/applications/components/application-summary/application-summary.tsx]: By boilerplate code, do you mean the getters/setters or something else? I have removed the getters and setters in the latest commit, but if it's something else, I'll work on fixing that too.

[the remark this answers is not in the document]
Comment (alexmt) [reply ui/src/app/applications/components/application-summary/application-summary.tsx]: Looking

**Review by alexmt — APPROVED** [review]
LGTM

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
I think we are good to merge

**Comment by alexec** [comment]
good work @AdityaGupta1 !

**Pull request merged**